ADR/0027-availability-probe-framework.md

```diff
@@ -0,0 +1,125 @@
+# 27. Availability Probe Framework
+
+* Date 2023-07-06
+
+## Status
+
+Accepted
+
+## Context
+
+As an AppStudio developer building functionality for the platform, I want to be able to
+easily visualize and comprehend the stability and availability of deployed systems in
+order to inform and influence future work towards improving the overall system
+reliability.
+  
+Such indication should tell us the overall up time of AppStudio with respect to services
+under the control of AppStudio developers.
+
+AppStudio is defined to be available at a given moment if all of its components are
+reporting to be available at that given moment.
+
+A component is defined to be available at a given moment if all of its availability
+probes are reporting to be available at that given moment.
+
+A convention is required for providing the availability of a probe.
+
+Once this is in place, those indicators can be aggregated in order to report the overall
+availability of AppStudio.
+
+## Decision
+
+Probes' availability will be provided as a Prometheus metric. The metric will contain
+metric labels to allow differentiating between the different probes. When exported out
+of the cluster of origin, additional labels will be attached to the metric to mark the
+cluster in which the metric was generated.
+
+### Delivery Method
+
+The availability Prometheus metric will be computed for each component based on the exit
+status of the latest execution of the CronJobs evaluating the component's availability.
+Developer teams will provide the CronJobs' implementation. The job results will be
+translated a standardized Prometheus metric in a unified fashion (i.e. developers
+will not be required to provide the translation mechanism).
```

Comment:
```suggestion
translated to a standardized Prometheus metric in a unified fashion (i.e. developers
will not be required to provide the translation mechanism).
```

This statement doesn't seem clear enough to me. What does it mean that the translation will happen in a unified fashion? Developers won't be required to provide the translation mechanism because they are adhering to some standard? If so, then just state something along those lines -- by adhering to a specific standard (be precise about the standard, link to it below), results will be aggregated into a standardized Prometheus metric to report on availability.